File: digital_wallet_frontend/src/services/adminService.ts
import api from './api';
import type { User, PaginatedResponse, UserDetail, AdminTransaction, TransactionStats } from '../types';

export function listUsers(search: string, page: number, size: number): Promise<PaginatedResponse<User>> {
  return api.get('/admin/users', { params: { search, page, size } }).then((res) => res.data);
}

export function getUserDetail(id: number): Promise<UserDetail> {
  return api.get(`/admin/users/${id}`).then((res) => res.data);
}

export function disableUser(id: number): Promise<{ status: string; message: string }> {
  return api.put(`/admin/users/${id}/disable`).then((res) => res.data);
}

export function enableUser(id: number): Promise<{ status: string; message: string }> {
  return api.put(`/admin/users/${id}/enable`).then((res) => res.data);
}

export function listTransactions(
  username: string,
  from: string,
  to: string,
  page: number,
  size: number
): Promise<PaginatedResponse<AdminTransaction>> {
  return api.get('/admin/transactions', { params: { username: username || undefined, from: from || undefined, to: to || undefined, page, size } }).then((res) => res.data);
}

export function getTransactionStats(from: string, to: string): Promise<TransactionStats> {
  return api.get('/admin/transactions/stats', { params: { from: from || undefined, to: to || undefined } }).then((res) => res.data);
}

File: digital_wallet/src/main/java/com/digital_wallet/service/AdminService.java
package com.digital_wallet.service;

import com.digital_wallet.model.dto.AdminTransactionDTO;
import com.digital_wallet.model.dto.PaginatedResponse;
import com.digital_wallet.model.dto.TransactionStatsDTO;
import com.digital_wallet.model.dto.UserDTO;
import com.digital_wallet.model.dto.UserDetailDTO;

public interface AdminService {
    PaginatedResponse<UserDTO> listUsers(String search, int page, int size);

    UserDetailDTO getUserDetail(Long userId);

    void disableUser(Long userId);

    void enableUser(Long userId);

    PaginatedResponse<AdminTransactionDTO> listTransactions(String username, String from, String to, int page, int size);

    TransactionStatsDTO getTransactionStats(String from, String to);
}
